### DISCUSSION TITLE
MD(sve)X to LaTeX!

### DISCUSSION THREAD

**Discussion by riziles** [executablebooks/meta#939] [Show and tell] [answered]
@rowanc1 , you probably already thought of this application, but how about using myst-js to convert MDX (or MDsveX) to LaTeX? Check out this example:

 <https://flourishing-hotteok-608239.netlify.app/mdsvex01>

**Comment by rowanc1** [comment, marked as answer]
Super cool, looks like you have got the latex/pdf features working locally at least? For a full working example of a `myst-demo` you can checkout the code here:
https://github.com/executablebooks/myst-theme/blob/main/packages/myst-demo/src/index.tsx

We are planning _lots_ of cleanups over the next month to make this easier, but if you want to dive in all of the pieces are referenced from that file and end up looking something like the sandbox here:
https://myst-tools.org/sandbox

The `myst-to-react` repo in [`@myst-theme`](https://github.com/executablebooks/myst-theme/tree/main/packages/myst-to-react) is essentially very very similar to mdx, and you can use different react components to override and render specific directives.

If you need other pointers around the place, let me know! We need to improve the developer docs a lot (as well as, well, all the docs!), so these questions help us prioritize various pieces. :)

**Reply by riziles** [reply]
> you can use different react components to override and render specific directives.

Oh, wow. I didn't realize that. I've seen the Myst React repo, but the Readme file implies that all it does is render Myst markdown. I didn't realize it supported executable JavaScript cells.

**Reply by rowanc1** [reply]
The `myst-to-react` repo is quite composable. How it works at the moment is with a [provider](https://github.com/executablebooks/myst-theme/blob/main/packages/site/src/pages/Root.tsx#L49) that wraps an article, for example. That can pass any sort of renderers in (here is what is used in a [default site](https://github.com/executablebooks/myst-theme/blob/main/packages/site/src/components/renderers.ts#L6)). The myst-to-react uses `myst-providers` to give you control over links, which are the main component that often requires special attention in applications (e.g. in Remix, Nextjs, or in [JupyterLab](https://github.com/executablebooks/jupyterlab-myst)), you can also override any component there with a specific React node (e.g. an [iframe](https://github.com/executablebooks/myst-theme/blob/main/packages/myst-to-react/src/iframe.tsx)).

I hope that gives you maybe a few more hints on how to use the tools in a stand-alone way. If you do something - would love to see it -- and can also continue to answer questions or give you pointers to code!

Currently our big push is to make [directives and roles easier to extend](https://github.com/executablebooks/mystjs/issues/181), at that point we will hopefully be getting back into the react/theme a bit more to both document it and make it a bit easier to use. Any feedback you have would be appreciated!

**Reply by riziles** [reply]
This all sounds amazing! I still think there might be interest from people who have already written something in MDX or MDsveX and are interested in exporting to a beautiful well formatted PDF. I basically did nothing to accommodate MyST aside from adding the front matter and a backup screenshot of the chart. I was thinking about using ViteTest to automatically freeze components, but it sounds like MyST JS is more extensible than I realized. Looking forward to learning more.

**Reply by rowanc1** [reply]
I think there is a rich space to explore on how to best connect MyST to the MDX community, and probably a few different paths to do it! Happy to do whatever I can to help you bridge these two communities. 🚀

**Reply by riziles** [reply]
@rowanc1 , I've spent some more time going through the docs, and if the functionality you are describing is documented somewhere, I sure as heck can't find it.

**Reply by rowanc1** [reply]
Hi @riziles, the developer docs on the libraries are pretty much non-existent, we have been focusing on the command-line documentation over the last few weeks. We certainly want to make that side more accessible, but it is going to take some time to pull the docs together!!

Is there something specific you are looking for, or a more general orientation?

**Reply by riziles** [reply]
Just trying to figure out how to put executable JavaScript into a MyST markdown file.
